nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,341 @@
+"""
+region_signal_extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from skimage.feature import peak_local_max
+from skimage.segmentation import random_walker
+
+from nilearn import plotting
+from nilearn.input_data import NiftiMasker, NiftiMapsMasker
+from nilearn._utils import check_niimg, check_niimg_3d, check_niimg_4d
+from nilearn._utils.extmath import fast_abs_percentile
+from nilearn.image import iter_img, new_img_like
+from nilearn.image.image import _smooth_array
+from nilearn._utils.niimg_conversions import concat_niimgs
+from nilearn._utils.compat import _basestring
+
+
+def extract_regions(map_data, affine, label_data, min_size):
+    """ Extract each region from the connected components
+        data where the data is already assigned with a
+        unique label point.
```

Comment:
I agree its not clear. What I mean is I am trying to tell that each voxel is assigned with unique label point of voxels in that particular region.